internal/list.go

```diff
@@ -11,3 +33,320 @@ func newListProxy(client *HazelcastClient, serviceName *string, name *string) (*
 	}
 	return &ListProxy{parSpecProxy}, nil
 }
+
+func (list *ListProxy) Add(element interface{}) (changed bool, err error) {
+	if element == nil {
+		return false, core.NewHazelcastNilPointerError(NIL_VALUE_IS_NOT_ALLOWED, nil)
+	}
+	valueData, err := list.ToData(element)
+	if err != nil {
+		return false, err
+	}
+	request := ListAddEncodeRequest(list.name, valueData)
+	responseMessage, err := list.Invoke(request)
+	if err != nil {
+		return false, err
+	}
+	changed = ListAddDecodeResponse(responseMessage).Response
+	return changed, nil
+}
+
+func (list *ListProxy) AddAt(index int32, element interface{}) (err error) {
+	if element == nil {
+		return core.NewHazelcastNilPointerError(NIL_VALUE_IS_NOT_ALLOWED, nil)
+	}
+	valueData, err := list.ToData(element)
+	if err != nil {
+		return err
+	}
+	request := ListAddWithIndexEncodeRequest(list.name, index, valueData)
+	_, err = list.Invoke(request)
+	return err
+}
+
+func (list *ListProxy) AddAll(elements []interface{}) (changed bool, err error) {
+	if len(elements) == 0 {
+		return false, core.NewHazelcastNilPointerError(NIL_SLICE_IS_NOT_ALLOWED, nil)
+	}
+	itemsData := make([]*Data, len(elements))
+	for index, item := range elements {
+		itemData, err := list.ToData(item)
+		if err != nil {
+			return false, err
+		}
+		itemsData[index] = itemData
+	}
+	request := ListAddAllEncodeRequest(list.name, itemsData)
+	responseMessage, err := list.Invoke(request)
+	if err != nil {
+		return false, err
+	}
+	changed = ListAddAllDecodeResponse(responseMessage).Response
+	return changed, err
+
+}
+
+func (list *ListProxy) AddAllAt(index int32, elements []interface{}) (changed bool, err error) {
+	if len(elements) == 0 {
+		return false, core.NewHazelcastNilPointerError(NIL_SLICE_IS_NOT_ALLOWED, nil)
+	}
+	elementsData := make([]*Data, len(elements))
+	for index, element := range elements {
+		itemData, err := list.ToData(element)
+		if err != nil {
+			return false, err
+		}
+		elementsData[index] = itemData
+	}
+	request := ListAddAllWithIndexEncodeRequest(list.name, index, elementsData)
+	responseMessage, err := list.Invoke(request)
+	if err != nil {
+		return false, err
+	}
+	changed = ListAddAllWithIndexDecodeResponse(responseMessage).Response
+	return changed, err
+}
+
+func (list *ListProxy) AddItemListener(listener interface{}, includeValue bool) (registrationID *string, err error) {
```

Comment:
`eventHandler` and `onItemEvent` clutters this method. Can you extract them out of this method?